ios/brave-ios/Sources/AIChat/Components/SpeechRecognition/AIChatSpeechRecognitionView.swift

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Foundation
+import SwiftUI
+import SpeechRecognition
+import AVFoundation
+
+struct AIChatSpeechRecognitionView: View {
+  @ObservedObject
+  var speechRecognizer: SpeechRecognizer
+  
+  @Binding
+  var isVoiceEntryPresented: Bool
+  
+  @Binding
+  var isNoMicrophonePermissionPresented: Bool
+  
+  @Binding
+  var recognizedText: String
+  
+  var body: some View {
+    SpeechToTextInputContentView(
+      isPresented: $isVoiceEntryPresented,
+      dismissAction: {
+        isVoiceEntryPresented = false
+      },
+      speechModel: speechRecognizer,
+      disclaimer: Strings.AIChat.speechRecognizerDisclaimer
+    )
+    .alert(isPresented: $isNoMicrophonePermissionPresented) {
+      Alert(
+        title: Text(Strings.AIChat.microphonePermissionAlertTitle),
+        message: Text(Strings.AIChat.microphonePermissionAlertDescription),
+        primaryButton: Alert.Button.default(
+          Text(Strings.settings),
+          action: {
+            let url = URL(string: UIApplication.openSettingsURLString)!
+            UIApplication.shared.open(url, options: [:], completionHandler: nil)
+          }
+        ),
+        secondaryButton: Alert.Button.cancel(Text(Strings.CancelString))
+      )
+    }
+    .onReceive(speechRecognizer.$finalizedRecognition, perform: { recognition in
```

Comment:
This doesnt need to be `onReceive` anymore as you are observing the speech recognizer directly. If `finalizedRecognition` is Published you can use `onChange` instead